cardano-api/src/Cardano/Api/MetaData.hs

```diff
@@ -120,14 +129,23 @@ jsonToMetadataValue  av =
           Nothing -> Api.TxMetaList <$> traverse jsonToMetadataValue xs
           Just ys -> Right (Api.TxMetaMap ys)
       Aeson.Object kvs ->
-        Api.TxMetaMap . List.sortOn fst <$> traverse convertPair (HashMap.toList kvs)
+        if HashMap.keys kvs /= ["hex"]
+          then Api.TxMetaMap . List.sortOn fst <$> traverse convertPair (HashMap.toList kvs)
+          else
+            case HashMap.elems kvs of
+              [Aeson.String bs] ->
+                case Base16.decode (Text.encodeUtf8 bs) of
+                  (bs2, "") -> Right $ TxMetaBytes bs2
+                  (_, bsf) -> Left $ ConversionErrBadBytes bsf
+              what -> Left $ ConversionErrExpected "[Aeson.String x]" (show what)
```

Comment:
Probably should show `kvs` here instead of `what`, because `what` is a list of values and `kvs` is a list of key/values.

When we get an error, we need to keys to figure out why `"hex"` is not the only key.

Reply:
Fixed.